gio/src/settings.rs

```diff
@@ -23,6 +35,68 @@ impl<O: IsA<Settings>> SettingsExtManual for O {
     fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError> {
         self.set_value(key, &value.to_variant())
     }
+
+    fn bind_with_mapping<P: IsA<glib::Object>>(
+        &self,
+        key: &str,
+        object: &P,
+        property: &str,
+        flags: SettingsBindFlags,
+        get_mapping: Option<Box<dyn Fn(&mut glib::Value, &glib::Variant) -> bool>>,
```

Comment:
Should this maybe be `Fn(&glib::Variant) -> Option<glib::Value>`?

Reply:
If the function pointer is `NULL`, gio uses the default mapping. So unless there's an explicit way to invoke that default mapping, we need to pass a `NULL` function pointer to gio to support this.

Reply:
> a new GVariant holding the data from value , or NULL in case of an error

Say the docs. So returning `NULL` instead of a variant on errors seems allowed.

Reply:
You would convert `None` into `false` and `Some(_)` into `true` and copy over the value to the pointer passed to the trampoline.

Reply:
I mean, the function would work exactly the same way but in a more Rust-y way with an optional return value instead of an out parameter and "success" boolean return value.

Reply:
Ah, I see. So it would be `get_mapping: Option<Box<dyn Fn(&glib::Variant) -> Option<glib::Value>>`? (In the absence of changing to use a builder or such.) That should be nicer to work with.

Reply:
Yeah that's what I meant

Reply:
Though it might be necessary to also pass a `Type`, since I think in general a mapping might work with multiple property types... If supporting that is important.

Reply:
How would that type be used and where?

Reply:
I'm guessing the default mapping is like that. It could be useful to define a mapping that could be used in several places, but wouldn't be relevant in most cases. Just worth noting.

I was wondering whether there's any issue setting the value to the wrong type, but it seems to print a helpful warning:

> GLib-GObject-WARNING **: 08:45:28.918: unable to set property 'active' of type 'gboolean' from value of type 'gchararray'

So it is probably fine to just change this to return `Option<Value>`.

Reply:
Though there's also the matter of symmetry with `get_mapping`, which takes a `glib::VariantType`.